### DISCUSSION TITLE
RFC: Self-contained MAUI app packaged for store distribution

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3449] [Q&A]
Over the next few months, I'll be taking an app we build with Oqtane as an eLearning experience and packaging it for the store(s).  This release of the app will be a static set of pages and published as a solid version of the app.  We'll try to localizes as much of the content we can into the app itself.  The Oqtane database itself is only used for page management, we offload the state and logins to external services. (B2C, Cosmos).

I've been contemplating the possibility of making the app as self-contained as possible.  I know this breaks the client/server nature of Oqtane and might not be possible.  Ideally a MAUI Hybrid app would have as much of the app as possible included in the app.

Here are some ideas for configuration

- default database - SQLite localized to the app
- render mode MAUI Hybrid
- change the client/services to call Server DLL directly instead of HTTP
- load images and other web assets in MAUI from local assets instead of server
Here is a video of the app, to give you an idea of what we are building.

https://github.com/oqtane/oqtane.framework/assets/311063/b8ba78d8-721f-42e8-9f5e-1dc4830be013

**Comment by sbwalker** [comment]
If you want the app to be as self-contained as possible, I am not sure why you would try to use Oqtane? It is not clear what Oqtane could provide in this scenario as all of the services are exposed via server endpoints. It might make more sense to create a standard Blazor Hybrid app for this scenario.

**Reply by markdav-is** [reply]
good point.  I'm looking at packaging this week, so we'll see how it goes.
